fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/RecordMetaData.java

```diff
@@ -303,6 +326,22 @@ public static RecordMetaData build(@Nonnull RecordMetaDataProto.MetaData proto)
         return RecordMetaData.newBuilder().setRecords(proto).getRecordMetaData();
     }
 
+    /**
+     * Factory method to deserialize a record meta-data proto. It assumes that the proto contains all of the dependencies
+     * and does not process extension options.
+     * @param proto the serialized proto message of the {@code RecordMetaData}
+     * @param enableIndexCounterBasedSubspaceKey use counter-based subspace keys for indexes
+     * @return the {@code RecordMetaData} object
+     */
+    @Nonnull
+    public static RecordMetaData build(@Nonnull RecordMetaDataProto.MetaData proto, boolean enableIndexCounterBasedSubspaceKey) {
```

Comment:
Is there a common use case where there's a `RecordMetaDataProto.MetaData` that doesn't have the subspace keys you want already? They would have to be unset in the proto for this to have an effect, too.

Reply:
I was thinking of dynamic messages that are extended using a RecordMetaDataProto.MetaData.Builder and addIndexesBuilder. E.g., similar to this test:

https://github.com/FoundationDB/fdb-record-layer/blob/2c3b2db7d2fbfb34ea143ec62e7be38e621701c9/fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/metadata/MetaDataProtoTest.java#L279

But I don't know how common this is. Is it?

Reply:
> > I'd also be okay with something that threw an error if `index_counter_based_subspace_keys` was set to `true` but any of the existing indexes were missing subspace keys.
> 
> Are you proposing we allow `index_counter_based_subspace_keys` = true only when none of the existing indexes have explicit subspace keys? It seems a fine approach to me.

I think the opposite, actually--that you can only enable counter-based subspace keys if *all* indexes have explicit subspace keys. That way you don't accidentally change the implicit key for some index. (To be clear, I'm not sure throwing an error is necessary, just saying when I think throwing an error should be done if we went with that approach.)

> Regardless of what we decide to do, I think what I have is good too. In my opinion, enabling counter-based subspace keys is an explicit enough change for existing clients to make its subsequences predictable.
> 
> > Also, if this field is in the meta-data proto, should the behavior be controlled by that rather than by the parameter the user sets?
> 
> As I mentioned above, I was considering dynamic messages. I will drop it per your other suggestion.

What do you mean by dynamic messages here?